lib/dataset/bins.cpp

```diff
@@ -309,11 +322,22 @@ Variable histogram(const Variable &data, const Variable &binEdges) {
   // 1-D histogramming provided that the input has multiple bins along
   // `hist_dim`.
   const Dim dummy = Dim::InternalHistogram;
-  if (indices.dims().contains(hist_dim))
+  const auto nbin = binEdges.dims()[hist_dim] - 1;
+  if (indices.dims().contains(hist_dim)) {
+    // With large existing dim matching the new dim, we would create a large
+    // intermediate histogrammed result, which leads to performance and memory
+    // issues. This is a suboptimal (since it concatenates first) but simple way
+    // to avoid the problem.
+    // Note that unless there are additional dimensions, the following nested
+    // call to histogram will currently not use threading.
+    if (indices.dims().volume() * nbin > 100000000) // about 1 GByte
+      return histogram(concatenate(data, hist_dim), binEdges);
```

Comment:
Was it possible to histogram in this case before? (On a beefy workstation) If so, did this change impact run time significantly?

Reply:
See my earlier post in the PR thread. TL;DR No, there were cases where it crashed. Yes some runtimes are impacted significantly.

Reply:
But only negatively in one case. I think this is acceptable.